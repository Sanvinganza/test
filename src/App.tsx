import { observer } from "mobx-react-lite";
import { MovieList } from "./components/MoviesList";
// import { MovieTable } from "./components/MoviesTable/TableMovies";

const App = observer(() => {
  return (
    <>
      <MovieList />
      {/* <MovieTable /> */}
    </>
  );
});

export default App;
