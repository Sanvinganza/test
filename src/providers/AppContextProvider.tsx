import { FC } from "react";
import { moviesContext } from "../hooks/useAppContext";
import { useLocalObservable } from "mobx-react-lite";
import { moviesState } from "../state";

export const MoviesContextProvider: FC<any> = ({ children }) => {
  const moviesStatelocalObservable = useLocalObservable(() => moviesState);

  return (
    <moviesContext.Provider value={moviesStatelocalObservable}>
      {children}
    </moviesContext.Provider>
  );
};
