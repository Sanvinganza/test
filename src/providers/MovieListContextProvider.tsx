import { FC } from "react";
import { movieListContext } from "../components/MoviesList/hooks/useMovieListContext";
import { useLocalObservable } from "mobx-react-lite";
import { movieListState } from "../states/stateMovieList";

export const MovieListContextProvider: FC<any> = ({ children }) => {
  const movieListStatelocalObservable = useLocalObservable(
    () => movieListState
  );
  return (
    <movieListContext.Provider
      value={movieListStatelocalObservable}
      children={children}
    />
  );
};
