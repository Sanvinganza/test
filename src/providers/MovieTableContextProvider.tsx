import { FC } from "react";
import { useLocalObservable } from "mobx-react-lite";
import { movieTableState } from "../states/stateMovieTable";
import { movieTableContext } from "../components/MoviesList/hooks/useMovieTableContext";

export const MovieTableContextProvider: FC<any> = ({ children }) => {
  const movieTableStatelocalObservable = useLocalObservable(
    () => movieTableState
  );

  return (
    <movieTableContext.Provider value={movieTableStatelocalObservable}>
      {children}
    </movieTableContext.Provider>
  );
};
