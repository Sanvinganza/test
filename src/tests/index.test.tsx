import { beforeEach, describe, expect, test } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import App from "../App";
import { MoviesContextProvider } from "../providers/MovieListContextProvider";

describe("App", () => {
  beforeEach(() => {
    const { asFragment } = render(
      <MoviesContextProvider>
        <App />
      </MoviesContextProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("После нажатия кнопки отображается элемент", () => {
    const button = screen.getByTestId("button-change-id");

    fireEvent.click(button);
    fireEvent.click(button);
    fireEvent.click(button);

    expect(screen.getAllByText("Harry").length).toEqual(3);
    expect(screen.getAllByText("Lorem impsom foues")).toBeDefined();
  });
});
