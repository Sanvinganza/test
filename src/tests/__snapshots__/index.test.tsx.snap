// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > После нажатия кнопки отображается элемент 1`] = `
<DocumentFragment>
  <h1>
    Films :
  </h1>
  <br />
  <ul>
    <ol>
      <h2>
        DR.Strange
      </h2>
      <h4>
        Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quisquam, magni/
      </h4>
    </ol>
    <ol>
      <h2>
        Harry Potter
      </h2>
      <h4>
        Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quisquam, magni/
      </h4>
    </ol>
    <ol>
      <h2>
        Star Wars
      </h2>
      <h4>
        Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quisquam, magni/
      </h4>
    </ol>
  </ul>
  <button
    data-testid="button-change-id"
  >
    add Movie "Harry"
  </button>
  <button
    data-testid="button-filter-id"
  >
    Filter Movies
  </button>
</DocumentFragment>
`;
